### DISCUSSION TITLE
GIT LFS Support / Dealing with large files

### DISCUSSION THREAD

**Discussion by wnwright** [alshedivat/al-folio#2242] [Q&A]
Hello! I've been trying to upload large files to my website (in access of 150-200 MB) and I am inevitably running into the standard github file size limit... I have attempted to get around this by using Git LFS, but it seems that the docker image doesn't support it....

Anyone run into similar issues with large files? How did you get around it?

**Comment by wnwright** [comment]
Okay, I actually got LFS to play nice but now I am dealing with additional byte requirements:

checkout@v4 is failing:

`error: RPC failed; HTTP 400 curl 22 The requested URL returned error: 400`

A little digging has show that this can be remedied with:
`git config --global http.version HTTP/1.1'
 
'git config --global http.postBuffer 157286400`

However, this would require changing jame's action, right? Is there an easier way? Creating an additional action executing these commands before the Deploy step unfortunately doesn't help.

**Comment by george-gca** [comment]
If you check the [Github pages doc](https://docs.github.com/en/pages/getting-started-with-github-pages/about-github-pages) you will see:

- Github Pages source repositories have a recommended limit of 1 GB. For more information, see "About large files on GitHub"
- Published GitHub Pages sites may be no larger than 1 GB.
- GitHub Pages deployments will timeout if they take longer than 10 minutes.

So I don't know if it is possible to (and even if they would want to) support Git LFS for Github pages.

**Reply by wnwright** [reply]
Fair, but I am positive I am still under the 1GB limit in my case

**Reply by george-gca** [reply]
Actually I don't think you need to change James' action. If you look at the [deploy script](https://github.com/alshedivat/al-folio/blob/master/.github/workflows/deploy.yml), it is the last step of the deploy, which is basically copy the `_site/` directory to GitHub pages.

Checking [actions/checkout](https://github.com/actions/checkout) documentation, I found this:

```yml
# Whether to download Git-LFS files
# Default: false
lfs: ''
```

Probably all you need is to set it to true.

**Reply by wnwright** [reply]
Good find! Unfortunately, changing my deploy action still doesn't get rid of the HTTP error, even when adding the requisite git config commands to the aforementioned action... Should I be adding those commands elsewhere?

**Reply by george-gca** [reply]
In which step does this error occur? Also, have you looked at the other examples in the [actions/checkout](https://github.com/actions/checkout) documentation? There is [an example](https://github.com/actions/checkout?tab=readme-ov-file#push-a-commit-using-the-built-in-token) where they use `git config` there.
